File: weather-ui/src/app/core/cidade.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

import { Cidade } from './model';
import { environment } from './../../environments/environment';

@Injectable()
export class CidadeService {

  private cidadeUrl: string;

  constructor(private http: HttpClient) {
    this.cidadeUrl = `${environment.apiUrl}/cidades`;
  }

  // Método respónsavel por consumir api http que busca todas as cidades
  public listar(): Promise<Cidade[]> {
    return this.http.get<Cidade[]>(`${this.cidadeUrl}`).toPromise();
  }

  // Método respónsavel por consumir api http que busca a cidade que está syncronizando os dados
  public buscarCidadeSync(): Promise<Cidade> {
    return this.http.get<Cidade>(`${this.cidadeUrl}/sync`).toPromise();
  }

  // Método respónsavel por consumir api http que alterar a ciddade que está syncronizando os dados
  public AlterarCidadeSync(codigo: number): Promise<any> {
    return this.http.put(`${this.cidadeUrl}/${codigo}/ativo`, null).toPromise();
  }

}

File: weather-api/src/main/java/br/com/exemplo/weather/api/service/CidadeService.java
package br.com.exemplo.weather.api.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.exemplo.weather.api.model.Cidade;
import br.com.exemplo.weather.api.repository.CidadeRepository;

@Service
public class CidadeService {

	@Autowired
	private CidadeRepository cidadeRepository;

	/**
	 * Método respónsavel por alterar a cidade que está syncronizando os dados
	 * @param codigo
	 * @return Cidade
	 */
	public Cidade alterarCidadeSync(Long codigo) {

		Cidade cidadeAtiva = cidadeRepository.findByAtivoIsTrue();		
		Cidade cidade = cidadeRepository.findById(codigo).get();
		
		cidadeAtiva.setAtivo(false);
		cidade.setAtivo(true);

		cidadeRepository.save(cidadeAtiva);
		cidadeRepository.save(cidade);
		
		return cidade;
	}
	
}
